src/helpers.js

```diff
@@ -11,24 +11,39 @@
  * {one: {two: "bla"}}), `"value"` if the same key already exists swith a
  * different value, or `false`.
  */
-function dotPathToHash(entry, target = {}, options = {}) {
+function dotPathToHash(locale, entry, target = {}, options = {}) {
   let path = entry.key
   if (options.suffix || options.suffix === 0) {
     path += `_${options.suffix}`
   }
 
   const separator = options.separator || '.'
-  let newValue = entry.defaultValue || options.value || ''
-
-  if (options.skipDefaultValues) {
+  const key = entry.key.substring(
```

Comment:
This feels complicated, I believe we could accomplish the same thing with making only `defaultValue` be a function. Then I'd do something like:

```js
if (typeof options.defaultValue === 'function') {
  // run it
}
else {
  // the current logic
}
```

Reply:
I believe there is a use-case for making at least `skipDefaultValues` a function: since `entry.defaultValue` has higher precedence than `options.defaultValue`, there is still no way to skip the default value for entries that have a specified default value for only some locales (unless `skipDefaultValues` can also be a function).